orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -750,7 +750,7 @@ IMPLEMENT_GRADIENT_BUILDER(GetConvGradient) {
     if (IsGradientRequiredForSrcNodeInput(i)) {
       outputs.push_back(GI(i));
     } else {
-      outputs.push_back(ArgDef("", nullptr));
+      outputs.push_back(IA("ConvInput_" + I(i).name));
```

Comment:
Derek had a fix for this. The gradient for inputs are made optional, so this change is probably not needed anymore.

Reply:
Maybe we can keep my change here as it reads better then empty string.

Reply:
Just for my education: will it still be treated as an optional output if there's a non-empty string there? (If so, I'm happy with putting a more descriptive string in the graph.)

Reply:
@mrry , I had a quick test and indeed empty string can skip the output calculation while non-empty string cannot. I've rollbacked the change. Thanks!

Reply:
Thanks for testing this! I wonder if this means there are other places in the code base where we give names to unused outputs and end up doing wasted computation :).